ogr/services/forgejo/release.py

```diff
@@ -1,36 +1,145 @@
 # Copyright Contributors to the Packit project.
 # SPDX-License-Identifier: MIT
 
-from typing import Optional
+import datetime
+from functools import cached_property
+from typing import Any, Optional
 
+from pyforgejo import PyforgejoApi
+
+from ogr.abstract import GitTag as AbstractGitTag
 from ogr.abstract import Release
-from ogr.services import forgejo as ogr_forgejo
+from ogr.abstract import Release as AbstractRelease
+
+
+class ForgejoGitTag(AbstractGitTag):
+    """
+    Concrete implementation of GitTag for Forgejo
+    """
+
+    def __init__(self, name: str, commit_sha: str) -> None:
+        self.name = name
+        self.commit_sha = commit_sha
+
+    def __str__(self) -> str:
+        return f"GitTag(name={self.name}, commit_sha={self.commit_sha})"
+
+
+class ForgejoRelease(AbstractRelease):
+    """
+    Concrete implementation of Release for Forgejo.
+    """
+
+    @cached_property
+    def api(self) -> PyforgejoApi:
```

Comment:
we should be reusing the authenticated API client from the `ForgejoProject` class